### PR TITLE
Refactor of boundary system

### PR DESCRIPTION
**Pull request by StephenHodgson**
# XRTK - Mixed Reality Toolkit Pull Request

## Overview

This PR refactors the Boundary System as defined in #623 

## Breaking Changes

- Removed `IMixedRealityBoundaryHandler`
- `IMixedRealityBoundarySystem` no longer implements `IMixedRealityEventSource`
- Removed `IMixedRealityBoundarySystem.FloorScale`
- Removed `IMixedRealityBoundarySystem.FloorPhysicsLayer`
- Removed `IMixedRealityBoundarySystem.FloorMaterial`
- Removed `IMixedRealityBoundarySystem.ShowPlayArea`
- Removed `IMixedRealityBoundarySystem.PlayAreaMaterial `
- Removed `IMixedRealityBoundarySystem.PlayAreaPhysicsLayer`
- Removed `IMixedRealityBoundarySystem.ShowTrackedArea`
- Removed `IMixedRealityBoundarySystem.TrackedAreaMaterial`
- Removed `IMixedRealityBoundarySystem.TrackedAreaPhysicsLayer`
- Removed `IMixedRealityBoundarySystem.BoundaryWallMaterial`
- Removed `IMixedRealityBoundarySystem.BoundaryWallsPhysicsLayer`
- Removed `IMixedRealityBoundarySystem.BoundaryCeilingMaterial`
- Removed `IMixedRealityBoundarySystem.CeilingPhysicsLayer`
- Removed `IMixedRealityBoundarySystem.GetFloorVisualization`
- Removed `IMixedRealityBoundarySystem.GetPlayAreaVisualization`
- Removed `IMixedRealityBoundarySystem.GetTrackedAreaVisualization`
- Removed `IMixedRealityBoundarySystem.GetBoundaryWallVisualization`
- Added `IMixedRealityBoundarySystem.BoundaryProximityAlert`
- Added `IMixedRealityBoundarySystem.TrackedObjects`
- Added `IMixedRealityBoundarySystem.BoundaryDataProvider`
- Added `IMixedRealityBoundarySystem.IsVisible`
- Added `IMixedRealityBoundarySystem.IsConfigured`
- Added `IMixedRealityBoundarySystem.ShowBoundary`
- Added `IMixedRealityBoundarySystem.RegisterTrackedObject`
- Added `IMixedRealityBoundarySystem.UnregisterTrackedObject`
- Renamed `IMixedRealityBoundarySystem.Bounds` -> `IMixedRealityBoundarySystem.BoundaryBounds`
- Renamed `IMixedRealityBoundarySystem.Contains` -> `IMixedRealityBoundarySystem.IsInsideBoundary`
- Renamed `IMixedRealityBoundarySystem.ShowBoundaryWalls` -> `IMixedRealityBoundarySystem.ShowWalls`
- Renamed `IMixedRealityBoundarySystem.ShowBoundaryCeiling` -> `IMixedRealityBoundarySystem.ShowCeiling`

## Submodule Changes

- https://github.com/XRTK/Examples/pull/12
- https://github.com/XRTK/WindowsMixedReality/pull/86

### CODE REVIEW

**Comment by FejZa** [comment]
I can't see where the new `ProximityAlert` event is being raised.

**Comment by StephenHodgson** [comment]
Ah seems like I only raise it [once place so far](https://github.com/XRTK/XRTK-Core/pull/624/files#diff-6920b862995855712f229feaf85c2311R329). I think I stopped cause I didn't have any platform data providers implemented yet.

File: XRTK-Core/Packages/com.xrtk.core/Runtime/Services/BoundarySystem/MixedRealityBoundarySystem.cs [hunk 1]
```diff
@@ -28,553 +26,497 @@ public class MixedRealityBoundarySystem : BaseEventSystem, IMixedRealityBoundary
...
+                switch (trackedObjectStatus.Value)
                 {
-                    currentTrackedAreaObject.layer = trackedAreaPhysicsLayer;
+                    case ProximityAlert.Clear:
+                        if (!IsInsideBoundary(trackedObjectStatus.Key.transform.position))
+                        {
+                            BoundaryProximityAlert?.Invoke(trackedObjectStatus.Key, ProximityAlert.Touch);
+                        }
+                        break;
+                    case ProximityAlert.Touch:
```
Comment (SimonDarksideJ) [inline XRTK-Core/Packages/com.xrtk.core/Runtime/Services/BoundarySystem/MixedRealityBoundarySystem.cs:332]: Not implemented?
Can we have comments on these if they are meant to be empty.  Else remove

**Comment by FejZa** [comment]
Ah sorry it was my mistake. The diffs in the file were too many so I had to explicitly click "load diffs". During my first pass I skipped that file by mistake.

Good work so far. Let's wait for the submodule PRs / data providers to see the whole picture.

File: XRTK-Core/Packages/com.xrtk.core/Runtime/Services/BoundarySystem/MixedRealityBoundarySystem.cs (hunk 1 above)
Comment (StephenHodgson) [reply XRTK-Core/Packages/com.xrtk.core/Runtime/Services/BoundarySystem/MixedRealityBoundarySystem.cs:332]: I'll finish. I must have somehow over looked this by accident.

**Comment by StephenHodgson** [comment]
> Good work so far. Let's wait for the submodule PRs / data providers to see the whole picture.

WMR data provider is ready, but I didn't include it in the PR bc I wanna wait until our native API push to implement

**Comment by StephenHodgson** [comment]
I updated this PR. Should be ready to review again.

File: azure_pipelines/stages/unity-build.yml [hunk 2]
```diff
@@ -5,6 +5,7 @@ stages:
   jobs:
     - job:
       displayName: ${{ format('{0}', parameters.image) }}
+      timeoutInMinutes: 120
```
Comment (FejZa) [inline azure_pipelines/stages/unity-build.yml:8]: Seems unrelated but I am fine with leaving it in.

Comment (StephenHodgson) [reply azure_pipelines/stages/unity-build.yml:8]: Yeah I kept hitting the default timeout which was 60 min do I doubled it.

Although I did find more errors with our build scripts I need to address. I might actually just pull those out.

**Comment by StephenHodgson** [comment]
BTW the re-writer error is harmless. BUT our Windows64 build went from 7 min to 42 min 👀 

You guys see anything that would make that happen?

File: azure_pipelines/stages/unity-build.yml (hunk 2 above)
Comment (StephenHodgson) [reply azure_pipelines/stages/unity-build.yml:8]: https://github.com/XRTK/XRTK-Core/pull/640

**Comment by SimonDarksideJ** [comment]
Is this still not ready as the dependant PR's are still in draft?

**Comment by StephenHodgson** [comment]
It's a breaking change. The wmr submodule or cannot be reviewed until this PR is merged and a preview sent out.

**Pull request merged**